### ISSUE TITLE
If repository is marked as release only, use-releases will not be used to check for releases

### ISSUE THREAD

**Issue by JasonWThompson** [mojohaus/versions-maven-plugin#72]
Consider a pom with the following repository defined:

```
        <repository>
            <id>internal</id>
            <name>My Internal Release Repository</name>
            <url>http://repo.example.corp/internal/repo</url>
            <releases>
                <enabled>true</enabled>
            </releases>
            <snapshots>
                <enabled>false</enabled>
            </snapshots>
        </repository>

```

When executing the following command, the internal repo will be ignored.

`mvn versions:use-releases`

Only by enabling snapshots will the versions plugin choose to update the snapshots to the releases found in the internal repo. The expected behavior is that the versions plugin would do the exact opposite and know that it can find releases in this repo.

**Comment by khmarbaise** [comment]
For such cases you should `use-latest-releases` goal instead.
The docs of the `use-releases` goal:
`Replaces any -SNAPSHOT versions with the corresponding release version (if it has been released).`
which can only be considered if a SNAPSHOT repository exists for this...

**Comment by JasonWThompson** [comment]
I don't think I've explained myself well. I am aware of what this plugin should do. This is why I filed the bug a year ago. If my pom has 1.0-SNAPSHOT as a dependency and I release that dependency, then after running mvn versions:use-releases against the pom that contains said dependency should replace the 1.0-SNAPSHOT guy with 1.0. And it does exactly that as long as I do not provide hints as to which repo is which. When **I do provide the hints** then it **does not do as it describes**.

I'll try to come up with a patch and submit a pull request.
